### DISCUSSION TITLE
Multiple tests on different accounts

### DISCUSSION THREAD

**Discussion by Rocky-04** [DogsTailFarmer/martin-binance#49] [Q&A] [answered]
May I inquire if it is possible to run multiple tests on different Binance accounts, each with its unique set of configurations, using a single device?

**Comment by DogsTailFarmer** [comment, marked as answer]
Yes, no limits. [There](https://github.com/DogsTailFarmer/martin-binance/discussions/35#discussioncomment-4008208) some answers which may be useful.

**Comment by adamantmm** [comment]
Yes — one host, multiple accounts works fine if each bot process gets its own config + API keys and you isolate state directories (logs, SQLite/JSON caches, Telegram session if used).

What usually bites people:

- **rate limits** — two accounts on the same IP still share Binance weight buckets per key, but websocket connection caps and IP bans are host-level
- **shared files** — if two instances point at the same data path, grids overwrite each other
- **clock / NTP** — less dramatic on spot, but signed REST errors show up fast when the box drifts

I run one systemd unit (or docker compose service) per account with a dedicated working directory and env file; that keeps configs honest without needing multiple machines.
